src/graph.c

```diff
@@ -0,0 +1,392 @@
+/* Copyright 2016 Outscale SAS
+ *
+ * This file is part of Butterfly.
+ *
+ * Butterfly is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published
+ * by the Free Software Foundation.
+ *
+ * Butterfly is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Butterfly.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <glib.h>
+#include <packetgraph/packetgraph.h>
+#include "brick-int.h"
+#include "graph-int.h"
+
+struct pg_graph *pg_graph_new(const char *name, struct pg_brick *explore)
+{
+	struct pg_error *error = NULL;
+	struct pg_graph *ret = g_new(struct pg_graph, 1);
+
+	ret->name = g_strdup(name);
+	ret->all = g_hash_table_new(g_str_hash, g_str_equal);
```

Comment:
I think we should use that: https://developer.gnome.org/glib/2.30/glib-Hash-Tables.html#g-hash-table-new-full
This will allow use to set destroy callback